src/common/transformations/src/transformations/convert_precision.cpp

```diff
@@ -470,17 +523,33 @@ bool fuse_type_to_eye_v9(const std::shared_ptr<ngraph::Node>& node, const precis
     return false;
 }
 
-bool fuse_type_to_parameter(const std::shared_ptr<ngraph::Node>& node, const precisions_map& precisions) {
+bool fuse_type_to_parameter(const std::shared_ptr<ngraph::Node>& node,
+                            const precisions_map& precisions,
+                            bool convert_input_precision) {
     auto it = precisions.find(node->get_output_element_type(0));
     if (it == precisions.end())
         return false;
+    bool changed = false;
     const auto& to = it->second;
     if (auto param = ov::as_type_ptr<opset4::Parameter>(node)) {
-        param->set_element_type(to);
-        param->validate_and_infer_types();
-        return true;
+        if (convert_input_precision) {
+            param->set_element_type(to);
+            param->validate_and_infer_types();
+            changed = true;
+        } else {
+            auto param_consumers = param->output(0).get_target_inputs();
+            auto convert = std::make_shared<opset4::Convert>(param, to);
+            for (auto& input : param_consumers) {
+                const auto consumer = input.get_node();
+                if (ov::is_type<ov::op::v0::Result>(consumer) || ov::is_type<ov::op::v0::Convert>(consumer)) {
```

Comment:
what about Sink operations (Assign op)?

Reply:
I don't think we should handle them in ConvertPrecision. For example if we'd like keep Assign original precision (e.g. f64) its ReadValue counterpart will already have its input precision changed to f32 and we'd get a mismatch.